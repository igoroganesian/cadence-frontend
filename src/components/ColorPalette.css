.color-palette {
  border-radius: 10px;
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  justify-items: center;
  padding: 1em;
}

.color-palette-circle {
  box-sizing: border-box;
  width: 3em;
  height: 3em;
  border-radius: 50%;
  margin: 5px;
  cursor: pointer;
}

.selected {
  border: 4px solid rgb(79, 79, 79);
}

@media (max-width: 570px) {
  /* .color-palette {
    grid-template-columns: repeat(3, 1fr);
  } */
  .color-palette-circle {
    width: 2.5em;
    height: 2.5em;
  }
}