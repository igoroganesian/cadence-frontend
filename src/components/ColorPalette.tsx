import { ColorPaletteProps } from "../types";
import './ColorPalette.css';

const ColorPalette = ({ selectedColor, onSelectColor }: ColorPaletteProps) => {
    const colors = [
        '#636363', '#33FF57', '#3357FF', '#FFFF33', '#FF33FF',
        '#33FFFF', '#FF5733', '#C70039', '#900C3F', '#581845',
        '#FFC300', '#DAF7A6', '#FFC0CB', '#800080', '#008080',
        '#808000', '#000080', '#DC143C'
    ];

    return (
        <div className="color-palette">
            {colors.map(color => (
                <div
                    key={color}
                    data-testid="color-palette-circle"
                    className={`color-palette-circle ${selectedColor === color ? 'selected' : ''}`}
                    style={{ backgroundColor: color }}
                    onClick={() => {
                        onSelectColor(color);
                        console.log(color)
                    }}
                />
            ))}
        </div>
    );
};

export default ColorPalette;