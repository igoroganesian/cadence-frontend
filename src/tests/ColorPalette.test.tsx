import '@testing-library/jest-dom';
import { render, fireEvent } from '@testing-library/react';
import ColorPalette from '../components/ColorPalette';

function hexToRgb(hex: string) {
    const r = parseInt(hex.slice(1, 3), 16);
    const g = parseInt(hex.slice(3, 5), 16);
    const b = parseInt(hex.slice(5, 7), 16);
    return `rgb(${r}, ${g}, ${b})`;
}

describe('ColorPalette Component', () => {
    const mockOnSelectColor = jest.fn();

    it('renders without crashing', () => {
        render(<ColorPalette selectedColor="#FFFF33" onSelectColor={mockOnSelectColor} />);
    });

    it('renders correctly', () => {
        const { asFragment } = render(<ColorPalette selectedColor="#FFFF33" onSelectColor={mockOnSelectColor} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the correct number of color circles', () => {
        const { getAllByTestId } = render(<ColorPalette selectedColor="#FFFF33" onSelectColor={mockOnSelectColor} />);
        const colorCircles = getAllByTestId('color-palette-circle');
        expect(colorCircles.length).toBe(18);
    });

    it('indicates the selected color', () => {
        const selectedColor = "#636363";
        const rgbSelectedColor = hexToRgb(selectedColor);
        const { container } = render(<ColorPalette selectedColor={selectedColor} onSelectColor={mockOnSelectColor} />);
        const colorCircles = container.querySelectorAll('.color-palette-circle');
        // Type errors don't impact JS runtime, but HTMLElement is necessary here to avoid a VSCode error
        const selectedCircle = Array.from(colorCircles).find(circle => (circle as HTMLElement).style.backgroundColor === rgbSelectedColor);

        expect(selectedCircle).toHaveClass('selected');
    });

    it('calls onSelectColor with the correct color when a circle is clicked', () => {
        const selectedColor = "#636363";
        const rgbSelectedColor = hexToRgb(selectedColor);

        const { container } = render(<ColorPalette selectedColor="#FFFF33" onSelectColor={mockOnSelectColor} />);
        const colorCircles = container.querySelectorAll('.color-palette-circle');
        const circleToClick = Array.from(colorCircles).find(circle =>
            (circle as HTMLElement).style.backgroundColor === rgbSelectedColor
        );

        if (circleToClick) {
            fireEvent.click(circleToClick);
        }

        expect(mockOnSelectColor).toHaveBeenCalledWith(selectedColor);
    });

});
