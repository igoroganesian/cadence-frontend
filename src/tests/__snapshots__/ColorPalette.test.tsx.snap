// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPalette Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="color-palette"
  >
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(99, 99, 99);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(51, 255, 87);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(51, 87, 255);"
    />
    <div
      class="color-palette-circle selected"
      data-testid="color-palette-circle"
      style="background-color: rgb(255, 255, 51);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(255, 51, 255);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(51, 255, 255);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(255, 87, 51);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(199, 0, 57);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(144, 12, 63);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(88, 24, 69);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(255, 195, 0);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(218, 247, 166);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(255, 192, 203);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(128, 0, 128);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(0, 128, 128);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(128, 128, 0);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(0, 0, 128);"
    />
    <div
      class="color-palette-circle "
      data-testid="color-palette-circle"
      style="background-color: rgb(220, 20, 60);"
    />
  </div>
</DocumentFragment>
`;
